### ISSUE TITLE
Pyright unable to match overloads for os.path.basename

### ISSUE THREAD

**Issue by kumaraditya303** [microsoft/pyright#2341]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
Pyright is unable to handle type overload matching for `os.path.basename`

**To Reproduce**

```py
from __future__ import annotations

import os
from pathlib import Path


def func(arg: str | Path | int) -> str:
    if isinstance(arg, (str, Path)):
        path = os.path.basename(arg)
    else:
        path = str(arg)
    return path

```

**Expected behavior**
No error

**Error**

```console 
No configuration file found.
stubPath D:\playwright-python\typings is not a valid directory.
Assuming Python platform Windows
Searching for source files
Found 1 source file
D:\playwright-python\main.py
  D:\playwright-python\main.py:9:16 - error: No overloads for "basename" match the provided arguments
    Argument types: (str | Path) (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos 
Completed in 0.672sec

```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

```
pyright 1.1.126
```

**Comment by erictraut** [comment]
This doesn't repro with the latest version of pyright.

Are you really using pyright version 1.1.126? The latest version is 1.1.171. Please upgrade to a more recent version.

**Comment by kumaraditya303** [comment]
Thanks it does not occur on the latest version.
